File: Python/Stage 3/S3_PIPE_01.py
def pipe(*fns):
    def apply(value):
        result = value
        for fn in fns:
            result = fn(result)
        return result

    return apply


add2 = lambda n: n + 2
mul3 = lambda n: n * 3
sub1 = lambda n: n - 1

pipeline = pipe(add2, mul3, sub1)
print(pipeline(4))

File: Js/Task 5/S3_PIPE_01.js
function pipe(...fns) {
    return x => fns.reduce((v, f) => f(v), x);
}

const add = x => x + 1;
const multiply = x => x * 2;

const result = pipe(add, multiply)(3);  // (3 + 1) * 2 = 8
console.log(result);